DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UITabBarRecorder.swift

```diff
@@ -7,21 +7,73 @@
 #if os(iOS)
 import UIKit
 
-internal struct UITabBarRecorder: NodeRecorder {
+internal final class UITabBarRecorder: NodeRecorder {
     let identifier = UUID()
+
+    private var currentlyProcessedTabbar: UITabBar? = nil
+    private lazy var subtreeViewRecorder: ViewTreeRecorder = {
+        ViewTreeRecorder(
+            nodeRecorders: [
+                UIImageViewRecorder(
+                    tintColorProvider: { imageView in
+                        guard let imageViewImage = imageView.image else {
+                            return nil
+                        }
+                        guard let tabBar = self.currentlyProcessedTabbar else {
+                            return imageView.tintColor
+                        }
+
+                        // Retrieve the tab bar item containing the imageView.
+                        let currentItemInSelectedState = tabBar.items?.first {
+                            let itemSelectedImage = $0.selectedImage
+
+                            // Important note when comparing the different tab bar items' icons:
+                            // our hypothesis is that each item uses a different image.
+                            return itemSelectedImage?.uniqueDescription == imageViewImage.uniqueDescription
+                        }
+
+                        // If the item is not selected,
+                        // return the unselectedItemTintColor,
+                        // or the default gray color if not set.
+                        if currentItemInSelectedState == nil || tabBar.selectedItem != currentItemInSelectedState {
+                            let unselectedColor = tabBar.unselectedItemTintColor ?? .lightGray.withAlphaComponent(0.5)
+                            return tabBar.unselectedItemTintColor ?? .systemGray.withAlphaComponent(0.5)
```

Comment:
suggestion/ In some other places we avoid referencing system colors directly. Instead we bridge them through [`SystemColors`](https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogSessionReplay/Sources/Recorder/Utilities/SystemColors.swift) utility that provides fallbacks for older OS-es. Might be used here for consistency and better look-up (it already defines `.systemGray`).

Reply:
I followed your suggestion ✅